x-pack/plugins/enterprise_search/public/applications/app_search/components/multi_input_rows/multi_input_rows.tsx

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { useEffect, useRef } from 'react';
+
+import { useValues, useActions } from 'kea';
+
+import { EuiButton, EuiButtonEmpty, EuiSpacer } from '@elastic/eui';
+
+import { CONTINUE_BUTTON_LABEL } from '../../../shared/constants';
+
+import {
+  ADD_VALUE_BUTTON_LABEL,
+  DELETE_VALUE_BUTTON_LABEL,
+  INPUT_ROW_PLACEHOLDER,
+} from './constants';
+import { InputRow } from './input_row';
+import { MultiInputRowsLogic } from './multi_input_rows_logic';
+import { filterEmptyValues } from './utils';
+
+interface Props {
+  values: string[];
+  onSubmit(values: string[]): void;
+  submitsOnChange?: boolean;
+  submitButtonText?: string;
+  addRowText?: string;
+  deleteRowLabel?: string;
+  inputPlaceholder?: string;
+}
+
+export const MultiInputRows: React.FC<Props> = ({
+  values: initialValues,
+  onSubmit,
+  submitsOnChange = false,
```

Comment:
API usage rambling: I could have potentially made this a 2nd `onChange` function instead of `submitsOnChange`, but I didn't really want to deal with the typing shenanigans of figuring out "either an onSubmit or an onChange function is required" so I figured this was simpler? 😅 

It's definitely still an option if we ever (for some reason) want _both_ an onChange and onSubmit though. If you strongly prefer an `onChange` fn (I think I like it the more I think about it as well), let me know!

Reply:
Yeah, this is interesting. Since this is essentially a form component (not a form), I wouldn't expect it to have anything other than an `onChange`.

I guess it's kind of like an uncontrolled component, in the sense that it's managing managing its own state and just announcing changes that are made.

I'm having a little trouble wrapping my head around it at this point, but maybe it will make more sense as I continue to review and see how we're actually using it.

Reply:
As a heads up I went ahead and make the change to both optional onSubmit and onChange for both component clarity & flexibility

https://github.com/elastic/kibana/pull/96881/commits/dadcbfdcf06f6d170905b88ba49d3d3be1673627